### ISSUE TITLE
Support for file destination

### ISSUE THREAD

**Issue by chopper** [Blooie/grunt-redis#1]
It would be useful to have an option to specify the complete destination key in Redis as opposed to only a prefix. This would make for more generalizable functionality

**Comment by mansona** [comment]
So the prefix can be of any size (as far as i'm aware) and it can be of any style i.e. doesn't have to have a : in it but that is recommended by Redis. 

I think this actually works using the official grunt file patterns (read here: http://gruntjs.com/configuring-tasks#files ). Have you tried defining the file object like that?

**Comment by chopper** [comment]
Sorry, I

**Comment by chopper** [comment]
Sorry, I should've added a bit more detail. I'm not concerned about the format and length of the prefix, but rather that the Redis key always includes (at least a part of) the file path and there is no option to directly specify the key. So there does not seem to be a way of uploading the file to, say, `app:releases:current', because it will append the filename (or optionally the full path), so the key ends up being 'app:releases:currentindex.html'

Looking at `tasks/redis.js`, it looks like the key is defined as follows

`var key = (options.prefix) + (options.fullPath ? filepath : path.basename(filepath));`

I agree though that following the official grunt file patterns would be best, so instead of introducing a `key` code, however currently `files.dest` seems to not utilized. I'd propose having the option to specify a `files.dest` property that lets you specify the exact key you would like the file to be uploaded to.

**Comment by chopper** [comment]
Actually, looking at the [Grunt File Patterns](http://gruntjs.com/configuring-tasks#files), it looks like the current setup follows a non-standard pattern. 

The current configuration is expected like this: 
```
task: { 
   files: { 
      src: [<source files>] 
   }
}
```
Note that according to the Grunt docs the compact form is not wrapped in the `files` hash. The format should rather be:
```
task: { 
   src: [<source files>]
}
```
Or, in the case of a single file upload with a specific destination:
```
task: { 
   files: { 
      src: <source file>
      dest: <source>
   }
}
```
Although given that this plugin could upload multiple files in one job, but always has a 1-to-1 mapping between `src` and `dest`, the following syntax should be supported, too:
```
task: { 
   files: { 
      files: [
        {src: 'src/file1.js', dest: 'key:for:file1'},
        {src: 'src/file2.js', dest: 'key:for:file2'},
      ],
   }
}
```
To summarize I think it would make sense to update this plugin to 1) follow grunt file conventions and 2) add support for the `dest` key
